Fixes coin machine footer padding issue

agustif opened the pull request:
## Description

- [x] Fixes padding in the bottom of the coin machine, So 'We love feedback' doesn't get on top of it.

**New stuff** ✨

* Adds bottom padding value to mainContent class in ColonyHomeLayout.css


<img width="872" alt="Screenshot 2021-11-24 at 01 50 23" src="https://user-images.githubusercontent.com/6601142/143152552-249f168d-dcd5-4f9f-9142-276d05572b25.png">


Resolves #2902

agustif commented:
Moved to not touching colonyLayout with the padding hack, instead removed height from grid class in CoinMachine css, and moved margin-top to margin adding also a margin-bottom of 20px to it

Screenshots at different zoom levels:
<img width="1440" alt="Screenshot 2021-11-24 at 15 44 09" src="https://user-images.githubusercontent.com/6601142/143261006-4ae87492-1982-4dbe-b7e5-cc6a4aa49307.png">
<img width="1440" alt="Screenshot 2021-11-24 at 15 43 40" src="https://user-images.githubusercontent.com/6601142/143261010-e177935b-8d3e-4b7a-a06b-d178f6a41c08.png">

<img width="1439" alt="Screenshot 2021-11-24 at 15 44 21" src="https://user-images.githubusercontent.com/6601142/143260996-4f9660b7-8711-4611-9828-d9c7ebf0523f.png">

chinins reviewed (changes requested):
There is problem with the implementation. Please, see my gif.

The coin machine should be the height of the screen minus the margins and content around. It should not overflow for more than the screen height. I think you'll have to adapt Raul's calculation of the height, not just remove it.

I think it would also be good to have margin-right of 20-30px for the whole coin-machine since we are on the subject of margins here.
![coin-machine-scroll](https://user-images.githubusercontent.com/34057551/143270887-aa660248-2961-4d54-8363-2e09f6943875.gif)

agustif commented:
> There is problem with the implementation. Please, see my gif.
> 
> The coin machine should be the height of the screen minus the margins and content around. It should not overflow for more than the screen height. I think you'll have to adapt Raul's calculation of the height, not just remove it.
> 
> I think it would also be good to have margin-right of 20-30px for the whole coin-machine since we are on the subject of margins here. 

Moved implementation to coin machine css, not touching height now, only adding proper marigins.

Added margin right of 30px

Can you re-review @chinins thank you!

chinins reviewed (changes requested):
There is no margin bottom on the coin machine which is one of the issues to solve here.
<img width="1280" alt="Screenshot 2021-11-26 at 17 19 21" src="https://user-images.githubusercontent.com/34057551/143608715-f3b0d92e-9b22-4075-9a71-e1440dbe4d4e.png">

agustif commented:
![image](https://user-images.githubusercontent.com/6601142/143612996-fd1a09a8-2d9b-4721-9578-362f511f8495.png)
![image](https://user-images.githubusercontent.com/6601142/143613089-26b8525a-6bf1-4b45-b66b-58c340e25c25.png)

@chinins thanks! Fixed now, 
Rebasing!

chinins commented on src/modules/dashboard/components/CoinMachine/CoinMachine.css:
```diff
@@ -1,5 +1,6 @@
 .main {
   display: block;
+  margin-bottom: 100px;
```
Why 100px? I think it's very wrong...

agustif replied:
I've moved to using width and height with calc, otherwise main class is smaller than it's child grid, and margin's don't get applied on lower resolutions?

Hope it's good now, let me know. Thank you

chinins commented on src/modules/dashboard/components/CoinMachine/CoinMachine.css:
```diff
@@ -19,7 +20,7 @@
 .grid {
   display: grid;
-  margin-top: 40px;
+  margin: 40px 0 20px;
```
I think this 20px margin is not applied, I think it's useless here.

agustif replied:
Removed it

chinins commented on src/modules/dashboard/components/CoinMachine/CoinMachine.css:
```diff
@@ -60,12 +61,14 @@
 .sales {
   composes: container;
-  min-height: 100%;
+  margin-bottom: 50px;
```
Why 50px? Specs shared by Karol says 30px. And why do we apply on one of the elements only and not on the larger container?

agustif replied:
Removed it, applying styles only now to main container

chinins commented on src/modules/dashboard/components/CoinMachine/CoinMachine.css:
```diff
@@ -60,12 +61,14 @@
 }
 
 .comments {
+  margin: 0 30px 50px 0;
```
Same here - why do we apply on one of the elements only and not on the larger container?

agustif replied:
Done!

chinins reviewed (changes requested):
Both the margin (of 50 px instead of 30px) and the implementation are incorrect...

rdig commented:
Please fix the issues pointed out by @chinins after which I'll review as well :+1:

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/JoinColony/colonyDapp?pullRequest=2957) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/JoinColony/colonyDapp?pullRequest=2957) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: agustif<br/>:x: iClon Z4 PRO<br/><hr/>**iClon Z4 PRO** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/JoinColony/colonyDapp?pullRequest=2957) it.</sub>

agustif commented:
> Please fix the issues pointed out by @chinins after which I'll review as well 👍

I moved implementation to main class css now.

I had to make it's height and width bigger due to it getting smaller than its coin machine grid contents. Let me know if this is not appropiate on how I should best handle this (the weird margin-top in main before was due to the same issue)

I also added a little margin on the left, so in smaller resoutions doesnt clash with the side menu Buy Tokens or Extensions.

chinins reviewed (approved):
It seems like a strange way to achieve the goal but it's working properly. I would like for @rdig  to have a look as well and check on a larger screen that it stretches all the way (it should but it still needs to be tested).

rdig reviewed (changes requested):
This is still not good. I shouldn't have a scrollbar to the side on any height. (but I do currently, notice the screenshots)

Both the "Previous Batches" and "Chat" should grow and shrink with the viewport, but not show  a srollbar 

![Screenshot from 2021-12-01 20-03-16](https://user-images.githubusercontent.com/1193222/144288802-2183a167-2603-4068-99c2-6e2165758d55.png)
![Screenshot from 2021-12-01 20-03-21](https://user-images.githubusercontent.com/1193222/144288809-ec8a72c7-5b82-4103-9531-3027204b6763.png)

rdig commented:
Also, why does this PR have 7 commits!!!! for such a small change ?

agustif commented:
> Also, why does this PR have 7 commits!!!! for such a small change ?

I tried two different approaches and didn't clean up the history, will do and squash 'em into one commit after fixing new feedback!

agustif commented:
Fixed scroll issue.
By adding a media query and only adding a marging of 20% (could also be a height: 120% but that didn't seem better) so in case of smaller than 1024px height viewports the padding still shows.

Also contained into 1 commit

Can you guys please re-review thx

rdig reviewed (approved):
I'm good now with all the changes, however I would argue that the 20% margin when the media query triggers is bit to agressive.

See the screenshots I posted below, as it adds a scroll bar a bit too early (look at when the window is half the size, that's when you see it)

On the other hand, you've spent too much time on this already, so this will do for now.

![Screenshot from 2021-12-02 10-11-05](https://user-images.githubusercontent.com/1193222/144382966-98e53264-e914-4329-9c27-fcac48cca18b.png)
![Screenshot from 2021-12-02 10-11-44](https://user-images.githubusercontent.com/1193222/144382969-c3856411-497e-4073-bc14-7e980b4c87dc.png)
![Screenshot from 2021-12-02 10-11-49](https://user-images.githubusercontent.com/1193222/144382970-0891fb94-e64a-4638-bf3f-803863708239.png)

chinins reviewed (approved):
Agree with Raul that 20% padding is not ideal but good enough for now.

rdig reviewed (approved):
Works better after the media query change
